src/Interpreters/Streaming/JoinStreamDescription.h

```diff
@@ -64,11 +68,20 @@ struct JoinStreamDescription
     ///
     /// SELECT * FROM left ASOF JOIN right
     ///    ON left.k = right.k AND left.version < right.version AND
-    ///       lag_behind(left.ts, right.ts, 20)
+    ///       lag_behind(20ms, left.ts, right.ts)
+    /// SETTINGS keep_versions=16;
+    ///
+    /// OR
+    ///
+    /// SELECT * FROM left ASOF JOIN right
+    ///    ON left.k = right.k AND left.version < right.version AND
+    ///       aligned_with(left.ts, right.ts)
     /// SETTINGS keep_versions=16;
     UInt64 keep_versions;
+    bool enable_join_alignment;
     Int64 latency_threshold;
-    String lag_column;
+    Int64 quiesce_threshold;
+    String timestamp_column;
```

Comment:
alignment_column

Reply:
alignment can be timestamp or an integral column ?

Reply:
For `interleave` alignment, we don't need an alignment column, how we are going to represent that ?

Reply:
`lag_behind(1)` can do the work, we will need introduce a lag unit / type. It may be timestamp milliseconds, or it may be a block like in `lag_behind(1)`

Reply:
agree it, I will remove `aligned_with`, it's not neccessary
lag_behind(1, key1, key2), means lag unit/type is same with key1 and key2.
lag_behind(1) means by default 1 chunk ?. (Probably not needed)

Reply:
yes, i was thinking similar things regarding these scenarios. 

**lag_behind(1, key1, key2)**, means lag unit/type is same with key1 and key2. <== I think we still need allow specifying the unit like 1ms ? Otherwise if key1 is nanoseconds precision, if users like to milliseconds alignment (most of the time, it will be the case), then user need specify something like **lag_behind(50000000, ts1, ts2)** which appears not quite convenient  and intuitive.

**lag_behind(1ms)** means lag 1ms based on processing time -> yes, it is the same as just setting `join_latency_threshold`.

**lag_behind(1)** means by 1 chunk. <== we still need this for interleave reading / consuming ?

Reply:
Personally i prefer for timestamp lag behind, the first argument is always interval like 10ms instead of a raw number like 10.